client/product-details/index.js

```diff
@@ -37,13 +49,23 @@ jQuery( function ( $ ) {
 			event,
 			variation
 		) {
-			const quantity = $( '.quantity input[type=number]' ).val();
-			const variationPrice =
-				productVariations[ variation.variation_id ].amount;
+			const quantityInput = $( '.quantity input[type=number]' );
```

Comment:
I believe, we shouldn't engare into updating the element if input is missing, or variation price and currency are not accessible.